dace/transformation/dataflow/streaming_memory.py

```diff
@@ -246,14 +436,48 @@ def apply(self, state: SDFGState, sdfg: SDFG) -> nodes.AccessNode:
 
             # Replace access node and memlet tree with one access
             if self.expr_index == 0:
-                replacement = state.add_read(name)
+                replacement = state.add_read(output_gearbox_name)
                 state.remove_edge(edge)
                 state.add_edge(replacement, edge.src_conn, edge.dst, edge.dst_conn, edge.data)
             else:
-                replacement = state.add_write(name)
+                replacement = state.add_write(input_gearbox_name)
                 state.remove_edge(edge)
                 state.add_edge(edge.src, edge.src_conn, replacement, edge.dst_conn, edge.data)
 
+        if self.use_memory_buffering:
+
+            arrname = str(self.access)
+            vector_size = int(self.memory_buffering_target_bytes / desc.dtype.bytes)
+
+            # Vectorize access to global array.
+            dtype = sdfg.arrays[arrname].dtype
+            sdfg.arrays[arrname].dtype = dtypes.vector(dtype, vector_size)
+            new_shape = list(sdfg.arrays[arrname].shape)
+            contigidx = sdfg.arrays[arrname].strides.index(1)
+            new_shape[contigidx] /= vector_size
+            try:
+                new_shape[contigidx] = int(new_shape[contigidx])
+            except TypeError:
+                pass
+            sdfg.arrays[arrname].shape = new_shape
+
+            # Change strides
+            new_strides: List = list(sdfg.arrays[arrname].strides)
+
+            for i in range(len(new_strides)):
+                if i == len(new_strides) - 1:  # Skip last dimension since it is always 1
+                    continue
+                new_strides[i] = new_strides[i] / vector_size
+            sdfg.arrays[arrname].strides = new_strides
+
+            # Change subset int the post state
+            for e in sdfg.states()[-1].edges():
```

Comment:
This is necessary because otherwise a memlet subset out-of-bounds exception will occur since the data container is now vectorized but the subset is still the old one.